.github/workflows/installer.yml

```diff
@@ -0,0 +1,133 @@
+name: Make single-file installers
+
+on:
+  release:
+    types: 
+      - released
+      - prereleased
+  workflow_dispatch:
+  pull_request:  # Remove this when merging this PR
+  schedule:
+    - cron: "0 0 * * 0"  # This cron job should maybe just live in openforcefield/status/?
+
+jobs:
+  test:
+    name: Building single file installer on ${{ matrix.cfg.os }}, Python ${{ matrix.python-version }}
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macOS-latest, ubuntu-latest]
+        python-version: [3.6, 3.7]
+
+    env:
+      CI_OS: ${{ matrix.os }}
+      PYVER: ${{ matrix.python-version }}
+
+    steps:
+    - uses: actions/checkout@v2
+      
+    - name: Wait for 2 hours if triggered by a release
+      if: (github.event_name == 'release') || (github.event_name == 'prerelease')
+      run:
+        sleep 7200
+
+    - uses: conda-incubator/setup-miniconda@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+        activate-environment: constructor
+        environment-file: devtools/conda-envs/installer.yaml
+        auto-activate-base: false
+
+    - name: Additional info about the build
+      shell: bash
+      run: |
+        uname -a
+        df -h
+        ulimit -a
+
+    - name: Environment Information
+      shell: bash -l {0}
+      run: |
+        conda info
+        conda list
+
+    - name: Prepare and run the constructor
+      shell: bash -l {0}
+      run: |
+        cd ..
+
+        # Find the tag of the last release (excluding RCs)
+        # TODO: Make this a variable that can be passed through from trigger and/or allow RC
+        export LATEST_TAG=$(git ls-remote --tags https://github.com/openforcefield/openforcefield.git | cut -f2 | grep -v "rc" | tail -1 | sed 's/refs\/tags\///')
+
+        echo $LATEST_TAG
+
+        git clone https://github.com/openforcefield/toolkit-installer-constructor
+        cd toolkit-installer-constructor
+
+        python ../openforcefield/devtools/scripts/build_cookiecutter_json.py $LATEST_TAG $PYVER $CI_OS
+        cp new_cookiecutter.json cookiecutter/cookiecutter.json
+
+        cat new_cookiecutter.json
+
+        python run.py
+
+        pwd
+        ls
+        ls build
+
+    - name: Upload tarball as artifact
+      uses: actions/upload-artifact@v2
+      with:
+        path: toolkit-installer-constructor/build/openforcefield-*.sh
+
```

Comment:
Something about this isn't working, and I'm strangely having a difficult time keeping track of paths while moving around in these scripts.  The logs each pretty much look like this:
```
Run actions/upload-artifact@v2
  with:
    path: toolkit-installer-constructor/build/openforcefield-*.sh
    name: artifact
    if-no-files-found: warn
  env:
    CI_OS: ubuntu-latest
    PYVER: 3.7
    CONDA_PKGS_DIR: /home/runner/conda_pkgs_dir
##[warning]No files were found with the provided path: toolkit-installer-constructor/build/openforcefield-*.sh. No artifacts will be uploaded.
```

In recent memory, the following paths have failed:
* `/build/openforcefield-*.sh`
* `toolkit-installer-constructor/build/openforcefield-*.sh`
* `../toolkit-installer-constructor/build/openforcefield-*.sh`

The wildcard is fine: https://github.com/actions/upload-artifact#upload-using-a-wildcard-pattern

Reply:
I think you're missing one level of hierarchy. Here's what the directory structure looks like after the SFI constuctor runs:
```
jeffreywagner@JW-MBP$ cd projects/OpenForceField/toolkit-installer-constructor/
jeffreywagner@JW-MBP$ ls
README.md       cookiecutter    run.py
build           environment.yml temp_miniconda
jeffreywagner@JW-MBP$ cd build/
jeffreywagner@JW-MBP$ ls
openforcefield-0.7.1-py3.6-linux-64 openforcefield-0.7.1-py3.7-linux-64
openforcefield-0.7.1-py3.6-osx-64   openforcefield-0.7.1-py3.7-osx-64
jeffreywagner@JW-MBP$ cd openforcefield-0.7.1-py3.6-osx-64/
jeffreywagner@JW-MBP$ ls
build.sh                                   post_install.sh
construct.yaml                             test
openforcefield-0.7.1-py36-MacOSX-x86_64.sh tmp
jeffreywagner@JW-MBP$ 
```

Reply:
You're right - I was missing a dir with the same name as the script
```
Successfully created '/Users/runner/work/openforcefield/toolkit-installer-constructor/build/openforcefield-0.7.1-py3.6-osx-64/openforcefield-0.7.1-py36-MacOSX-x86_64.sh'.
```
```
Successfully created '/home/runner/work/openforcefield/toolkit-installer-constructor/build/openforcefield-0.7.1-py3.6-linux-64/openforcefield-0.7.1-py36-Linux-x86_64.sh'.
```

Reply:
Huzzah! 

Also, each run of the constructor builds _all platforms_, which has started failing recently (pacakges built on mac won't work on linux). 

So we'll want to make sure that mac machines only upload mac artifacts and linux machines only upload linux artifacts. That said, we probably _don't_ need separate runs for py36 and 37. So the matrix could just be {mac, linux}.